src/clients/python/__init__.py

```diff
@@ -982,11 +982,13 @@ def _prepare_request(self, inputs, outputs,
         # An input's data may be specified as a list of numpy arrays,
         # or as a shared memory handle.
         for inp_name, inp in inputs.items():
-            if not isinstance(inp, (list, tuple)) and type(inp) != c_void_p:
+            if (not isinstance(inp, (list, tuple))) and (type(inp) != c_void_p):
                 _raise_error("input '" + inp_name +
                              "' values must be specified as a list of numpy arrays" \
                              " or as a single c_void_p representing the shared memory handle")
             if type(inp) != c_void_p:
+                if (len(inp) == 2) and (type(inp[0]) == c_void_p) and (isinstance(inp[1], (list, tuple))):
```

Comment:
Put a comment here explaining that this case is that allows you to skip further processing for this input.

Reply:
Done